packages/react/use-query-param/src/hooks/useQueryParam.ts

```diff
@@ -1,17 +1,18 @@
 /** @tossdocs-ignore */
+import { NextRouter } from 'next/router';
 import { useNextRouter } from './useNextRouter';
 
-interface Options<T> {
-  parser?: (val: string) => T;
+interface Options<TParsed> {
+  parser?: (value: NextRouter['query'][string]) => TParsed;
   suspense?: boolean;
 }
 
-export function useQueryParam<T = string>(name: string): T | undefined;
-export function useQueryParam<T = string>(name: string, options: Options<T>): T | undefined;
-export function useQueryParam<T = string>(name: string, options?: Options<T>) {
+export function useQueryParam(name: string): NextRouter['query'][string];
+export function useQueryParam<TParsed>(name: string, options: Options<TParsed>): TParsed;
```

Comment:
if options have parser field, ReturnType<typeof parser> is need to be return type of useQueryParam. so we can remove unnecessary undefined.